client/src/components/JurisdictionAdmin/BatchRoundDataEntry.tsx

```diff
@@ -0,0 +1,645 @@
+  isEditing: boolean
+  isRenamingSheet: boolean
+  isTotalsSheet?: boolean
+  savedResults: { [choiceId: string]: number }
+  saveResults: (e: BaseSyntheticEvent) => void
+  sheets: Partial<ArrayField<IBatchResultTallySheet, 'id'>>[]
+}
+
+const BatchResultTallySheet: React.FC<IBatchResultTallySheetProps> = ({
+  areResultsFinalized,
+  cancelSheetRename,
+  confirmSheetRename,
+  contest,
+  deleteSheet,
+  enableEditing,
+  formMethods,
+  index,
+  initiateSheetRename,
+  isEditing,
+  isRenamingSheet,
+  isTotalsSheet,
+  savedResults,
+  saveResults,
+  sheets,
+}) => {
+  const { errors, formState, getValues, register, trigger, watch } = formMethods
+
+  return (
+    <>
+      <BatchResultTallySheetTable>
+        <thead>
+          <tr>
+            <th>Choice</th>
+            <th>Votes</th>
+          </tr>
+        </thead>
+        <tbody>
+          {contest.choices.map(choice => (
+            <tr key={choice.id}>
+              <td>{choice.name}</td>
+              <td>
+                {isTotalsSheet ? (
+                  <span>
+                    {sum(
+                      sheets.map(
+                        (_, i) =>
+                          watch(
+                            `resultTallySheets[${i}].results[${choice.id}]`
+                          ) || 0
+                      )
+                    )}
+                  </span>
+                ) : (
+                  <>
+                    <input
+                      aria-label={`${choice.name} Votes`}
+                      className={classnames(
+                        Classes.INPUT,
+                        errors.resultTallySheets?.[index]?.results?.[
+                          choice.id
+                        ] && Classes.INTENT_DANGER,
+                        // Visually hide this input instead of completely unmounting it to avoid
+                        // interfering with react-hook-form's state management
+                        !isEditing &&
+                          !formState.isSubmitting &&
+                          HIDDEN_INPUT_CLASS_NAME
+                      )}
+                      defaultValue={`${sheets[index]?.results?.[choice.id] ||
+                        0}`}
+                      disabled={formState.isSubmitting}
+                      name={`resultTallySheets[${index}].results[${choice.id}]`}
+                      ref={register({
+                        min: 0,
+                        required: true,
+                        valueAsNumber: true,
+                      })}
+                      type="number"
+                    />
+                    {!isEditing && !formState.isSubmitting && (
+                      <span>{savedResults[choice.id] || 0}</span>
+                    )}
+                  </>
+                )}
+              </td>
+            </tr>
+          ))}
+        </tbody>
+      </BatchResultTallySheetTable>
+
+      <Actions>
+        {!isTotalsSheet && (
+          <SheetActions>
```

Comment:
I think the solution of having all the sheet actions at the bottom is workable. My only quibbles are that (1) it's a bit cluttered - there are a lot of controls shown (2) the sheet actions are far away from the sheet name (especially the rename action).

Two other ideas I have:
- Add a dropdown menu in the current tab name that has two options: Rename Sheet and Delete Sheet. The menu would be triggered by clicking a small icon button in the sheet name (similar styling to a browser tab delete button, but a diff icon - maybe a downward triangle). The rename option could either make the sheet name editable in place or show a modal.
- Add a dropdown menu to the "Add a sheet" button with similar functionality

Reply:
As discussed in person, went with the second approach ✅